test/verify/check-storage-resize

```diff
@@ -35,33 +35,71 @@ class TestStorage(StorageCase):
         b.wait_visible("#storage-detail")
-        self.content_tab_wait_in_info(1, 2, "Name", "FSYS")
+        self.content_row_wait_in_col(1, 2, "/dev/TEST/vol")
+
+        # ext4, grow while mounted, then shrink and expect unmount
 
-        mountpoint = self.mount_root + "/admin/FSYS"
+        self.content_tab_action(1, 2, "Format")
+        self.dialog({ "name": "FSYS-EXT4", "type": "ext4" })
+        self.content_tab_wait_in_info(1, 2, "Name", "FSYS-EXT4")
 
+        mountpoint = self.mount_root + "/admin/FSYS-EXT4"
         self.content_tab_action(1, 2, "Mount")
         self.content_tab_wait_in_info(1, 2, "Mounted At", mountpoint)
 
-        self.content_tab_info_action(1, 1, "Size")
+        self.content_tab_action(1, 1, "Grow")
         self.dialog({ "size": 400 })
 
         size = int(m.execute("df -k --output=size %s | tail -1" % mountpoint).strip())
         assert (size > 300000)
 
-        self.content_tab_info_action(1, 1, "Size")
-        self.dialog({ "size": 200 })
+        self.content_tab_action(1, 1, "Shrink")
+        self.dialog_wait_open()
+        b.wait_in_text("#dialog", "Proceeding will unmount all filesystems on it.")
+        self.dialog_set_val("size", 200)
+        self.dialog_apply()
+        self.dialog_wait_close()
 
-        # Shrinking a filesystem will unmount it
         self.content_tab_action(1, 2, "Mount")
         self.content_tab_wait_in_info(1, 2, "Mounted At", mountpoint)
 
         size = int(m.execute("df -k --output=size %s | tail -1" % mountpoint).strip())
         assert (size < 300000)
 
+        # xfs, grow while mounted, check that shrink is disabled
+
+        self.content_tab_action(1, 2, "Format")
+        self.dialog({ "name": "FSYS-XFS", "type": "xfs" })
+        self.content_tab_wait_in_info(1, 2, "Name", "FSYS-XFS")
+
+        mountpoint = self.mount_root + "/admin/FSYS-XFS"
+        self.content_tab_action(1, 2, "Mount")
+        self.content_tab_wait_in_info(1, 2, "Mounted At", mountpoint)
+
+        self.content_tab_action(1, 1, "Grow")
+        self.dialog({ "size": 400 })
+
+        size = int(m.execute("df -k --output=size %s | tail -1" % mountpoint).strip())
+        assert (size > 300000)
+
+        self.wait_content_tab_action_disabled(1, 1, "Shrink")
+
+        # Encrypted ext4, grow while mounted, then shrink and expect unmount
+
+        self.content_tab_action(1, 2, "Format")
+        self.dialog({ "name": "FSYS-LUKS", "type": "luks+ext4",
+                      "passphrase": "vainu-reku-toma-rolle-kaja",
+                      "passphrase2": "vainu-reku-toma-rolle-kaja" })
+        self.content_tab_wait_in_info(2, 1, "Name", "FSYS-LUKS")
+
+        # No version of UDisks supports resizing encrypted block
+        # devices yet.  Once we get such a version, the buttons will
+        # automatically be enabled and this test needs to be adapted.
+        self.wait_content_tab_action_disabled(1, 1, "Shrink")
+        self.wait_content_tab_action_disabled(1, 1, "Grow")
```

Comment:
Could this also test a file system other than ext4 and xfs? For these two there are hardcoded results of `query_fsys_info()`, but it would be good to get coverage for the `CanResize()` code path.

Reply:
Yes, very good point.  I added NTFS.